cmd/zoekt-sourcegraph-indexserver/index.go

```diff
@@ -321,6 +324,34 @@ func gitIndex(c gitIndexConfig, o *indexArgs, l sglog.Logger) error {
 		"-submodules=false",
 	}
 
+	// We store the document ranks as JSON in gitDir and add a flag to tell
+	// zoekt-git-index to rank based on the scores in the file, instead of ranking
+	// the files by itself.
+	if o.UseOfflineRanking {
+		args = append(args, "-offline_ranking")
+
+		r, err := sourcegraph.GetDocumentRanks(context.Background(), o.Name)
+		if err != nil {
+			return err
+		}
+
+		b, err := json.Marshal(r)
+		if err != nil {
+			return err
+		}
+
+		fd, err := os.Create(filepath.Join(gitDir, zoekt.DocumentScoresFile))
```

Comment:
it doesn't really seem we need to have zoekt.DocumentScoresFile as something exported. It looks private to this implementation (since you pass it in as a flag argument)

Edit: I see, I didn't notice this is a boolean flag. What do you think of making things more explicit and passing a path instead? This is minor design, either way is right just bringing up my initial expectation.

Reply:
I thought about that too, but opted against sending the file name because we might want to store repo scores in a separate file to keep the format simpole and I didn't want to send two file names. However, I am going to adopt your suggestion because it makes more sense right now. When we introduce repo ranks we can always reconsider.